packages/blocks/src/store/reducer.js

```diff
@@ -80,8 +80,19 @@ export const fallbackBlockName = createBlockNameSetterReducer( 'SET_FALLBACK_BLO
  * @return {Object} Updated state.
  */
 export function categories( state = DEFAULT_CATEGORIES, action ) {
-	if ( action.type === 'SET_CATEGORIES' ) {
-		return action.categories || [];
+	switch ( action.type ) {
+		case 'SET_CATEGORIES':
+			return action.categories || [];
+		case 'ADD_CATEGORY':
+			return [
```

Comment:
See also [Lodash's `_.uniqBy`](https://lodash.com/docs/4.17.10#uniqBy):

```js
const { title, slug } = action;
return uniqBy( [ ...state, { title, slug } ], 'slug' );
```

Reply:
I considered this too, but it will result in the original category being kept, rather than the new one overriding the original one - the latter is the expected behavior IMHO. 

An alternative would be to place `...state` after the new category in the array, but then the new category will be added to the beginning, which is unexpected (I expect the new category to go at the end of the array). 

What do you think?

Reply:
Interesting! I hadn't considered that. Considering order here is important, we should consider this case of adding one which already exists. One case which is not considered yet by the tests is if we add a category whose slug already exists _in the middle of_ (not at the end of) the array. As implemented here, it would shift the category with that slug to the end of the array. Is that expected? Or would it be an in-place replacement?